scripts/dep_update.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash -xe
+# Run pre and run hooks
+# Detect whether the repo has been modified
+# Create pr if required
+# Run post hook
+run_hook(){
+  if [[ -x "$1" ]]
+  then
+    $1 || exit 1
+  else
+    if [[ "$1" =~ .*/run ]]
+    then
+      echo "Run hook $1 not found."
+      exit 1
+    fi
+  fi
+}
+cd ${WORKSPACE}/repo
+run_hook "gating/update_dependencies/pre"
+run_hook "gating/update_dependencies/run"
+if [[ -z "$(git status -s)" ]]
+then
+  echo "Repo is clean, prep script made no changes to be comitted"
+else
+  echo "Repo is dirty, preparing to propose changes"
+  # split URL into array separated by '/'s
+  IFS='/ ' read -r -a url_split <<< "$URL"
+  # owner/org is whatever is between the second and third / in the url
+  owner="${url_split[3]}"
+  # repo is anything after owner/
+  repo=${URL#https://github.com/${owner}/}
+  ssh_url="git@github.com:${owner}/${repo}"
+  message="This change updates the branch dependencies with the repository
+modifications made by running the Jenkins hook in
+gating/update_dependencies"
+
+  # Get / Create Jira issue
+  jira_summary="Update ${repo}:${BRANCH} dependencies"
+  issue=$(python ${WORKSPACE}/rpc-gating/scripts/jirautils.py \
+        --user "${JIRA_USER}" \
+        --password "${JIRA_PASS}" \
+        create_issue \
+          --project "${JIRA_PROJECT_KEY}" \
+          --summary "${jira_summary}" \
+          --description "${message}" \
+          --type "${JIRA_ISSUE_TYPE}" \
+          --label RE_DEP_UPDATE \
+          --existing-issue-query "project = ${JIRA_PROJECT_KEY} AND resolution = Unresolved AND summary ~ \"${jira_summary}\""
```

Comment:
My understanding of this is that any new dependency update will be linked to an existing issue, if one exists, by updating the original pull request. This may end up reducing throughput, for example if the existing issue was in needs review and only waiting to be merged, updating the pull request would require all the tests to be rerun.

I would be inclined to only use an existing issue if it is in the backlog.

Reply:
Perhaps the right way to think about this is that if there is an unresolved card which is not in backlog then do not execute the dep update. The thinking there would be that in this situation the team may be dealing with fallout in the PR and the PR needs manual updates to get it into a working state - and preventing the dep update from running also prevents the PR being changed from the team's work to something else.

The only trouble I'm seeing here is that the column it's in could be named differently for each board. :/

Reply:
BACKLOG seems quite standard I'll change the query to `project = ${JIRA_PROJECT_KEY} AND status = Backlog AND summary ~ \"${jira_summary}\"`